Command request via external MQTT failed

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Hello there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

Please do not remove sections of this template when submitting your issue.  If the section is not relevant to your issue, leave the section but enter "n/a" for not applicable.

Note some sections are marked "REQUIRED" and must be provided.  Never mark required sections "n/a".

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug Report

### Affected Services [**REQUIRED**]
<!-- Can you pin-point one or more EdgeX services (Data, Metadata, Command, etc...) as the source of the bug? -->
<!-- ✍️edit: --> The issue is located in:  core-command


### Is this a regression?

<!-- Did this behavior work correctly in the previous version? -->
<!-- ✍️--> No


### Description and Minimal Reproduction [**REQUIRED**]
1. Run latest core-command
2. Send command request via external MQTT

## 🔥 Exception or Error
<pre><code>
level=DEBUG ts=2022-10-12T02:37:31.722552073Z app=core-command source=external.go:133 msg="Command request sent to internal MessageBus. Topic: edgex/device/command/request/device-simple/Simple-Device01/Xrotation/get, Correlation-id: 14a42ea6-c394-41c3-8bcd-a29b9f5e6835"
level=DEBUG ts=2022-10-12T02:37:31.724591496Z app=core-command source=internal.go:54 msg="Command response received on internal MessageBus. Topic: edgex/device/command/response/device-simple/Simple-Device01/Xrotation/get, Correlation-id: 14a42ea6-c394-41c3-8bcd-a29b9f5e6835 "
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x7b7b58]

goroutine 32 [running]:
github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging.publishMessage({0x0, 0x0}, {0xc0001226c0, 0x34}, 0x0?, 0x0?, {{0xc000022910, 0x49}, {0xc0003bc450, 0x24}, ...}, ...)
        github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging/external.go:145 +0x118
github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging.SubscribeCommandResponses.func1()
        github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging/internal.go:63 +0x40f
created by github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging.SubscribeCommandResponses
        github.com/edgexfoundry/edgex-go/internal/core/command/controller/messaging/internal.go:45 +0x50a

</code></pre>


## 🌍  Your Environment
**Deployment Environment:**
MacOS

**EdgeX Version [**REQUIRED**]:**
latest

**Anything else relevant?**

--- a/internal/core/command/controller/messaging/external.go
+++ b/internal/core/command/controller/messaging/external.go
@@ -15,6 +15,7 @@ import (
 	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+
 	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
 
 	"github.com/edgexfoundry/edgex-go/internal/core/command/container"
@@ -55,6 +56,7 @@ func OnConnectHandler(router MessagingRouter, dic *di.Container) mqtt.OnConnectH
 func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Container) mqtt.MessageHandler {
 	return func(client mqtt.Client, message mqtt.Message) {
 		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+		lc.Debugf("Received command query request from external message queue on topic '%s' with %d bytes", message.Topic(), len(message.Payload()))
 
 		requestEnvelope, err := types.NewMessageEnvelopeFromJSON(message.Payload())
 		if err != nil {
@@ -83,6 +85,8 @@ func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Co
 func commandRequestHandler(router MessagingRouter, dic *di.Container) mqtt.MessageHandler {
 	return func(client mqtt.Client, message mqtt.Message) {
 		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+		lc.Debugf("Received command request from external message queue on topic '%s' with %d bytes", message.Topic(), len(message.Payload()))
+
 		messageBusInfo := container.ConfigurationFrom(dic.Get).MessageQueue
 		qos := messageBusInfo.External.QoS
 		retain := messageBusInfo.External.Retain

--- a/internal/core/command/controller/messaging/internal.go
+++ b/internal/core/command/controller/messaging/internal.go
@@ -13,6 +13,7 @@ import (
 	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
 	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
 
 	"github.com/edgexfoundry/edgex-go/internal/core/command/container"
@@ -59,11 +60,13 @@ func SubscribeCommandResponses(ctx context.Context, router MessagingRouter, dic 
 					continue
 				}
 
+				// original request is from external MQTT
 				if external {
 					publishMessage(externalMQTT, responseTopic, qos, retain, msgEnvelope, lc)
 					continue
 				}
 
+				// original request is from internal MessageBus
 				err = messageBus.Publish(msgEnvelope, responseTopic)
 				if err != nil {
 					lc.Errorf("Could not publish to internal MessageBus topic '%s': %s", responseTopic, err.Error())

--- a/internal/core/command/main.go
+++ b/internal/core/command/main.go
@@ -94,6 +94,11 @@ func MessageBusBootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startup
 	router := messaging.NewMessagingRouter()
 
 	if configuration.RequireMessageBus {
+		if configuration.MessageQueue.External.Enabled {
+			if !handlers.NewExternalMQTT(messaging.OnConnectHandler(router, dic)).BootstrapHandler(ctx, wg, startupTimer, dic) {
+				return false
+			}
+		}
 		if !handlers.MessagingBootstrapHandler(ctx, wg, startupTimer, dic) {
 			return false
 		}
@@ -108,11 +113,6 @@ func MessageBusBootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startup
 		if err := messaging.SubscribeCommandQueryRequests(ctx, dic); err != nil {
 			lc.Errorf("Failed to subscribe command query request from internal message bus, %v", err)
 			return false
-		}
-		if configuration.MessageQueue.External.Enabled {
-			if !handlers.NewExternalMQTT(messaging.OnConnectHandler(router, dic)).BootstrapHandler(ctx, wg, startupTimer, dic) {
-				return false
-			}
 		}
 	}
 

